matrix-client.el

```diff
@@ -493,32 +507,73 @@ If MESSAGE is nil, clear existing message."
              (handler (a-get (oref con :event-handlers) type)))
     (funcall handler con room item)))
 
+(defun matrix-client--get-date-header (timestamp)
+  "Return position of appropriate date header in current buffer for TIMESTAMP.
+Creates a new header if necessary."
+  (cl-labels ((prev-header-pos () (matrix--prev-property-change (point) 'matrix-header-day-number))
```

Comment:
Oh, I've always wanted this and I never knew how to do it cleanly, I'll have to remember this :)

Reply:
Yeah, I don't see it used very often in elisp, but I think it can clarify and simplify code a lot.